### DISCUSSION TITLE
Help logging selected file name to the DNP

### DISCUSSION THREAD

**Discussion by jenningsb2** [chhoumann/quickadd#507] [Q&A] [answered]
I'm looking for the correct syntax for accessing the selected file here.

![image](https://github.com/chhoumann/quickadd/assets/29644050/92e99661-b565-45f9-9003-88b9a6c18495)

I'd like to do a second Capture in the macro that logs the selected file (and header) from the first capture.

Here is the first part of the macro:

![CleanShot 2023-05-27 at 11 33 47@2x](https://github.com/chhoumann/quickadd/assets/29644050/de0cf638-aca9-4651-9e4f-3dd091085ba8)

Here is what I am hoping to do with the second capture:

![CleanShot 2023-05-27 at 11 32 23@2x](https://github.com/chhoumann/quickadd/assets/29644050/34016f4f-b2ca-4d7c-b6f8-ed62956061a3)

**Comment by nicobako** [comment]
In your first macro function, you can store the _selected file_ in a variable as follows: 

```js
params.variables["selectedFile"] = selectedFile;
```

Then, in the capture, you can access the variable as follows:

`{{VALUE:selectedFile}}`

I've done something similar, and it works really well!

**Reply by jenningsb2** [reply]
Dumb question...where would I type that in?

**Reply by nicobako** [reply]
Not a dumb question.

You would put that into a "script".

However, if you're not comfortable using scripts then i would use an "inline script".

In the first step of your macro, think you could do

```
```js quickadd
// Get the selected file

// Store the selected file
this.variables["selectedFile"] = selectedFile;
```
```

Then, in the next steps if your macro the selected file should be available via `{{VALUE: selectedFile}}`.

**Reply by nicobako** [reply]
I guess what I'm trying to say is that i don't think there is a "format syntax" to do what you want to do. What you want to do is a little more sophisticated, and would require some code, but it can definitely be done with QuickAdd.

**Reply by nicobako** [reply]
If this still doesn't make sense, you can share the template file that is called from the first capture and I'll try and show you how to fix this.

**Reply by nicobako** [reply]
Okay, i went ahead and made a simple example.

---
First template 

```js quickadd
const input = await this.quickAddApi.inputPrompt("✍");
this.variables["input"] = input;
return `Input given: ${input}`;
```

Now i can reuse the value i stored in the inline script.
Input previously given: {{VALUE:input}}

---

Next caption in the macro

Input from first template: {{VALUE:input}}

---

Of course, you'll probably want to use a [suggester](https://quickadd.obsidian.guide/docs/QuickAddAPI#suggester-displayitems-string--value-string-index-number-arr-string--string-actualitems-string-promisestring) instead of an _input_ prompt.

**Reply by jenningsb2** [reply, marked as answer]
Thanks for your help @nicobako 

After _a lot_ of trial and error, I got the outcome I was hoping for!

```js
module.exports = async function listFiles(params) {
    const fileType = { 
        folder: "Meetings/Recurring",
    }
    // Get all files
    const allFiles = params.app.vault.getMarkdownFiles();

    // Filter files that start with 'Meetings/Recurring/'
    const files = allFiles.filter(file => file.path.startsWith(fileType.folder));

    // Use the suggester to select a file from the filteredFiles
    const notesDisplay = await params.quickAddApi.suggester(
        files.map(file => file.basename),
        files
    );

    // Assign the path of the selected file to a variable
    params.variables = {
        selectedFilePath: notesDisplay.path,
        selectedFileName: notesDisplay.basename
    };
};
```
